pkg/networkservice/mechanisms/srv6/client_test.go

```diff
@@ -0,0 +1,95 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package srv6_test
+
+import (
+	"context"
+	"io/ioutil"
+	"testing"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/networkservicemesh/sdk-vppagent/pkg/networkservice/mechanisms/srv6"
+	"github.com/networkservicemesh/sdk-vppagent/pkg/networkservice/utils/checks/testinterfaceappender"
+	"github.com/networkservicemesh/sdk-vppagent/pkg/networkservice/vppagent"
+
+	"github.com/networkservicemesh/sdk/pkg/networkservice/common/mechanisms/checkmechanism"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+	srv6_mechanism "github.com/networkservicemesh/api/pkg/api/networkservice/mechanisms/srv6"
+)
+
+func TestSrv6Client(t *testing.T) {
+	// Turn off log output
+	logrus.SetOutput(ioutil.Discard)
+	parameters := configureTestSRv6Parameters()
+	localInterfaceName := "client-ConnectionId"
+	testMechanism := configureTestSRv6Mechanism(parameters)
+	testRequest := &networkservice.NetworkServiceRequest{
+		Connection: &networkservice.Connection{
+			Id:        "ConnectionId",
+			Mechanism: testMechanism,
+		},
+	}
+	c := next.NewNetworkServiceClient(
+		testinterfaceappender.NewClient(),
+		srv6.NewClient(),
+	)
+	testConnToClose := testRequest.GetConnection()
+	suite.Run(t, checkmechanism.NewClientSuite(
+		c,
+		vppagent.WithConfig,
+		srv6_mechanism.MECHANISM,
+		func(t *testing.T, mechanism *networkservice.Mechanism) {
+			m := srv6_mechanism.ToMechanism(mechanism)
+			require.NotNil(t, m)
+		},
+		func(t *testing.T, ctx context.Context) {
+			conf := vppagent.Config(ctx)
+			require.NotNil(t, conf)
+			require.NotNil(t, conf.GetVppConfig())
+			require.NotNil(t, conf.GetLinuxConfig())
+			require.NotNil(t, conf.GetNetallocConfig())
+
+			// Basic Checks
+			vppConfig := conf.GetVppConfig()
+			vppInterfaces := vppConfig.GetInterfaces()
+			require.Greater(t, len(vppInterfaces), 0)
+			vppInterface := vppInterfaces[len(vppInterfaces)-1]
+			assert.NotNil(t, vppInterface)
+
+			// Checks that sets the correct configs
+			assert.Equal(t, expectedVppConfigSrv6Localsids(parameters, localInterfaceName), vppConfig.Srv6Localsids)
+			assert.Equal(t, expectedVppConfigSrv6Policies(parameters), vppConfig.Srv6Policies)
+			assert.Equal(t, expectedVppConfigSrv6Steerings(testRequest, parameters, localInterfaceName), vppConfig.Srv6Steerings)
+
+			// Make sure that the Request invokes the appendInterfaceConfig function with connect=true
+			if len(vppConfig.Vrfs) == 1 {
```

Comment:
should be used `require.Len(...)` ?

Reply:
Oh, I had written the incorrect comment. I mean we make sure with which connect flag (true for Request(...) and false for Close(...)) we invoke the appendInterfaceConfig(...) function because if it true then we append additional parameters to vppConfig(like &vpp_l3.VrfTable{...}, &vpp.Route{...} and &vpp.ARPEntry{...})

Reply:
Comment was fixed